src/time_integration/paired_explicit_runge_kutta/methods_PERK2.jl

```diff
@@ -0,0 +1,403 @@
+                                                            bS, c_end)
+
+    b1 = 1.0 - bS
+
+    return PairedExplicitRK2(num_stages, a_matrix, c, b1, bS, c_end)
+end
+
+# Constructor that calculates the coefficients with polynomial optimizer from a
+# semidiscretization
+function PairedExplicitRK2(num_stages, tspan, semi::AbstractSemidiscretization,
+                           verbose = false,
+                           bS = 1.0, c_end = 0.5)
+    eig_vals = eigvals(jacobian_ad_forward(semi))
+
+    a_matrix, c = compute_PairedExplicitRK2_butcher_tableau(num_stages,
+                                                            eig_vals, tspan,
+                                                            bS, c_end,
+                                                            verbose)
+
+    b1 = 1.0 - bS
+
+    return PairedExplicitRK2(num_stages, a_matrix, c, b1, bS, c_end)
+end
+
+# Constructor that calculates the coefficients with polynomial optimizer from a
+# list of eigenvalues
+function PairedExplicitRK2(num_stages, tspan, eig_vals::Vector{ComplexF64},
+                           verbose = false,
+                           bS = 1.0, c_end = 0.5)
+    a_matrix, c = compute_PairedExplicitRK2_butcher_tableau(num_stages,
+                                                            eig_vals, tspan,
+                                                            bS, c_end,
+                                                            verbose)
+
+    b1 = 1.0 - bS
+
+    return PairedExplicitRK2(num_stages, a_matrix, c, b1, bS, c_end)
+end
+
+# This struct is needed to fake https://github.com/SciML/OrdinaryDiffEq.jl/blob/0c2048a502101647ac35faabd80da8a5645beac7/src/integrators/type.jl#L1
+mutable struct PairedExplicitRKOptions{Callback}
+    callback::Callback # callbacks; used in Trixi
+    adaptive::Bool # whether the algorithm is adaptive; ignored
+    dtmax::Float64 # ignored
+    maxiters::Int # maximal number of time steps
+    tstops::Vector{Float64} # tstops from https://diffeq.sciml.ai/v6.8/basics/common_solver_opts/#Output-Control-1; ignored
+end
+
+function PairedExplicitRKOptions(callback, tspan; maxiters = typemax(Int), kwargs...)
+    PairedExplicitRKOptions{typeof(callback)}(callback, false, Inf, maxiters,
+                                              [last(tspan)])
+end
+
+abstract type PairedExplicitRK end
+abstract type AbstractPairedExplicitRKSingleIntegrator <: PairedExplicitRK end
+
+# This struct is needed to fake https://github.com/SciML/OrdinaryDiffEq.jl/blob/0c2048a502101647ac35faabd80da8a5645beac7/src/integrators/type.jl#L77
+# This implements the interface components described at
+# https://diffeq.sciml.ai/v6.8/basics/integrator/#Handing-Integrators-1
+# which are used in Trixi.
+mutable struct PairedExplicitRK2Integrator{RealT <: Real, uType, Params, Sol, F, Alg,
+                                           PairedExplicitRKOptions} <:
+               AbstractPairedExplicitRKSingleIntegrator
```

Comment:
This is so ugly it makes me weep 😭

But there's nothing you can do, it's our auto-formatter, thus feel free to ignore this comment 🙂